src/plugins/drawing/components/object-list-view.tsx

```diff
@@ -107,52 +110,84 @@ interface IObjectLineProps {
   setHoverObject: (id: string|null) => void
 }
 
-function ObjectLine({object, content, selection, setHoverObject}: IObjectLineProps) {
+const ObjectLine = observer(function ObjectLine({object, content, selection, setHoverObject}: IObjectLineProps) {
+
+  const {
+    attributes,
+    listeners,
+    setNodeRef,
+    transform,
+    transition,
+    active,
+    isDragging
+  } = useSortable({id: object.id});
 
   function handleHoverIn() {
+    if (active) return; // avoid flashes of highlight while dragging
     setHoverObject(object.id);
   }
 
   function handleHoverOut() {
     setHoverObject(null);
   }
 
-  function handleClick() {
-    content.setSelectedIds([object.id]);
+  // Select the clicked object, or add to existing selection with modifier key.
+  function handleClick(e: React.MouseEvent) {
+    if (e.shiftKey || e.metaKey) {
+      content.selectId(object.id);
```

Comment:
I'd expect this to deselect if it's already selected.

Reply:
That would be reasonable, but it would also be inconsistent with shift-clicking in the drawing canvas.  Right now regular click on objects toggles their selected status; shift- or command-clicking adds the object without any toggle. 

So if we want to be consistent we should actually either change this _and_ change click behavior in the canvas, or change how clicks without the modifier key work to be consistent with the canvas.